src/github.com/couchbase/sync_gateway/rest/api_test.go

```diff
@@ -1773,3 +1781,46 @@ func TestCreateTarget(t *testing.T) {
 	response = rt.sendRequest("PUT", "/foo/", "")
 	assertStatus(t, response, 403)
 }
+
+// Test for issue 758 - basic auth with stale session cookie
+func TestBasicAuthWithSessionCookie(t *testing.T) {
+
+	var rt restTester
+
+	// Create two users
+	response := rt.sendAdminRequest("PUT", "/db/_user/bernard", `{"name":"bernard", "password":"letmein", "admin_channels":["bernard"]}`)
+	assertStatus(t, response, 201)
+	response = rt.sendAdminRequest("PUT", "/db/_user/manny", `{"name":"manny", "password":"letmein","admin_channels":["manny"]}`)
+	assertStatus(t, response, 201)
+
+	// Create a session for the first user
+	response = rt.send(requestByUser("POST", "/db/_session", `{"name":"bernard", "password":"letmein"}`, "bernard"))
+	log.Println("response.Header()", response.Header())
+	assert.True(t, response.Header().Get("Set-Cookie") != "")
+
+	cookie := response.Header().Get("Set-Cookie")
+
+	// Create a doc as the first user, with session auth, channel-restricted to first user
+	reqHeaders := map[string]string{
+		"Cookie": cookie,
+	}
+	response = rt.sendRequestWithHeaders("PUT", "/db/doc1", `{"hi": "there", "channels":["bernard"]}`, reqHeaders)
```

Comment:
I think this test would be a lot easier to understand if you renamed:

* doc1 -> bernardsDoc
* doc2 -> mannysDoc